realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SyncedRealmTests.kt

```diff
@@ -232,6 +239,222 @@ class SyncedRealmTests {
         }
     }
 
+    @Test
+    fun embeddedObject_roundTrip() {
+        val user1: User = createNewUser()
+        val config1: SyncConfiguration = createDefaultConfig(user1, partitionValue)
+        val primaryKeyValue = UUID.randomUUID().toString()
+        Realm.getInstance(config1).use { realm ->
+            assertTrue(realm.isEmpty)
+
+            realm.executeTransaction {
+                realm.createObject<EmbeddedSimpleParent>(primaryKeyValue).let { parent ->
+                    realm.createEmbeddedObject<EmbeddedSimpleChild>(parent, "child")
+                }
+            }
+            realm.syncSession.uploadAllLocalChanges()
+
+            assertEquals(1, realm.where<EmbeddedSimpleParent>().count())
+            assertEquals(1, realm.where<EmbeddedSimpleChild>().count())
+        }
+
+        val user2: User = createNewUser()
+        val config2: SyncConfiguration = createDefaultConfig(user2, partitionValue)
+        Realm.getInstance(config2).use { realm ->
+            assertEquals(0, realm.where<EmbeddedSimpleParent>().count())
```

Comment:
Is this always true? Wouldn't the sync potentially already have fetched stuff.

Reply:
I thought of adding this because I was debugging something else a number of times and I found it curious that the local realm never had any objects at this point. I guess in all fairness that the synchronisation process could just be slow, so making this assumption might not be correct. I will delete it.